### DISCUSSION TITLE
How to convert <streamsync.core.BytesWrapper object at 0x1065ca890> to base64

### DISCUSSION THREAD

**Discussion by ManishAradwad** [writer/writer-framework#274] [Q&A] [answered]
I'm taking an image as a user input and I want to convert it into base64. How can I do that?

**Comment by ramedina86** [comment]
Hi, what are you trying to do? How are you getting the image from the user?

I believe you should only see `BytesWrapper` if you're trying to create an image (e.g. AI generated) and send it to the user.

**Reply by ManishAradwad** [reply]
I added a file input component first and then reading it as `state["profile_1"] = ss.pack_bytes(payload, mime_type="image/jpeg")`. The user should upload the image and then I want to use the base64 encoding of that image in my code.

**Reply by ManishAradwad** [reply]
`print(state["profile_1"])` gives me <streamsync.core.BytesWrapper object at 0x1065ca890>

**Reply by ramedina86** [reply, marked as answer]
`ss.pack_bytes()` is meant to take some data from the backend and then serialise it so that it can be shown in the frontend. For example, if you have some bytes generated by AI. But taking input is different.

Check out the stub from the _File Input_ component (by clicking on the help icon next to the event handler in Builder).

You get an array of the uploaded files. Each element of that array is a dict with `name` and `data`. The latter is a file-like object.

I've pasted it here for your convenience:

```py
def onchange_handler(state, payload):

	# An array of dictionaries is provided in the payload
	# The dictionaries have the properties name, type and data
    # The data property is a file-like object

    uploaded_files = payload
    for i, uploaded_file in enumerate(uploaded_files):
		name = uploaded_file.get("name")
        file_data = uploaded_file.get("data")
        with open(f"{name}-{i}.jpeg", "wb") as file_handle:
            file_handle.write(file_data)`.trim();
```

If for some reason you want to convert to base64 you'd do something like this:

```py
import base64

def handle_image_upload_and_print_base64(payload):
  uploaded_data = payload[0].get("data")
  print(base64.b64encode(uploaded_data.read()))
```

Please note that there's no need to convert to base64 (unless of course you need it for your own reasons).

**Reply by ManishAradwad** [reply]
Thanks @ramedina86 ! I'll try it
